app/scripts/background.js

```diff
@@ -394,7 +396,7 @@ function setupController (initState, initLangCode) {
         const origin = url.hostname
 
         remotePort.onMessage.addListener((msg) => {
-          if (msg.data && msg.data.method === 'eth_requestAccounts') {
+          if (msg.data && ['eth_requestAccounts', 'eth_sendTransaction'].includes(msg.data.method)) {
             requestAccountTabIds[origin] = tabId
```

Comment:
This stores the `tabId` for the _most recent_ tab to interact with MetaMask via either of these two methods. While this might be correct in most or all cases (at least so far), we should instead be able to find the `tabId` associated with specific requests instead, so that we can ensure we're redirecting the user back to the tab that initiated the specific confirmation they just finished.